Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnCancel.hs

```diff
@@ -121,9 +142,162 @@ tfFulfillments =
+            itemLocationIds = Nothing,
+            itemPaymentIds = Nothing,
+            itemPrice = tfItemPrice booking,
+            itemTags = Nothing
+          }
+      ]
+
+tfItemPrice :: DRB.Booking -> Maybe Spec.Price
+tfItemPrice booking =
+  Just
+    Spec.Price
+      { priceComputedValue = Nothing,
+        priceCurrency = Just "INR",
+        priceMaximumValue = Nothing,
+        priceMinimumValue = Nothing,
+        priceOfferedValue = Nothing,
+        priceValue = Just $ encodeToText booking.estimatedFare
+      }
+
+tfVehicle :: DRB.Booking -> Maybe Spec.Vehicle
+tfVehicle booking = do
+  let (category, variant) = BUtils.castVariant booking.vehicleVariant
+  Just $
+    Spec.Vehicle
+      { vehicleCategory = Just category,
+        vehicleVariant = Just variant,
+        vehicleColor = Nothing,
+        vehicleMake = Nothing,
+        vehicleModel = Nothing,
+        vehicleRegistration = Nothing
+      }
+
+tfAgent :: (CacheFlow m r, EsqDBFlow m r) => DRB.Booking -> m (Maybe Spec.Agent)
+tfAgent booking = do
+  driverQuote <- QDQ.findById (Id booking.quoteId) >>= fromMaybeM (QuoteNotFound booking.quoteId)
+  pure $
+    Just $
+      Spec.Agent
+        { agentContact = Nothing,
+          agentPerson =
+            Just
+              Spec.Person
+                { personId = Nothing,
+                  personImage = Nothing,
+                  personName = Just driverQuote.driverName,
+                  personTags = Nothing
+                }
+        }
+
+tfCustomer :: (CacheFlow m r, EsqDBFlow m r, EncFlow m r) => DRB.Booking -> m (Maybe Spec.Customer)
+tfCustomer booking = do
+  riderId <- booking.riderId & fromMaybeM (BookingFieldNotPresent "rider_id")
+  riderDetails <- runInReplica $ QRiderDetails.findById riderId >>= fromMaybeM (RiderDetailsNotFound riderId.getId)
+  customerPhoneNo <- decrypt riderDetails.mobileNumber
+  return $
+    Just $
+      Spec.Customer
+        { customerContact =
+            Just
+              Spec.Contact
+                { contactPhone = Just customerPhoneNo
+                },
+          customerPerson =
+            Just
+              Spec.Person
+                { personId = Nothing,
+                  personImage = Nothing,
+                  personName = booking.riderName,
+                  personTags = Nothing
+                }
+        }
+
 tfCancellation :: SBCR.CancellationSource -> Maybe Spec.Cancellation
 tfCancellation cancellationSource =
   Just $
     Spec.Cancellation
-      { cancellationCancelledBy = Just $ show cancellationSource
+      { cancellationCancelledBy = castCancellatonSource cancellationSource
       }
+  where
+    castCancellatonSource = \case
+      SBCR.ByUser -> Just (show Enums.CONSUMER)
+      SBCR.ByDriver -> Just (show Enums.PROVIDER)
+      _ -> Nothing -- if it is cancelled by any other source like by ByMerchant, ByAllocator or ByApplication then we are considering it as None.
```

Comment:
For others also send `PROVIDER`. From network perspective its provider who cancelled